components/transfer/transfer.component.ts

```diff
@@ -245,11 +245,8 @@ export class NzTransferComponent implements OnInit, OnChanges, OnDestroy {
 
   handleSelect(direction: TransferDirection, checked: boolean, item?: TransferItem): void {
     const list = this.getCheckedData(direction);
-    if (list.every(i => i.disabled)) {
-      this.updateOperationStatus(direction, 0);
-      return;
-    }
-    this.updateOperationStatus(direction, list.length);
+    const count = list.filter(i => !i.disabled).length;
+    this.updateOperationStatus(direction, count);
```

Comment:
Good Job! 
Notice that use `count` instead of `list.length`, so that the status of transfer button will be disabled if there is any checked item being disabled.
This issue also exists in antd.

Reply:
ref https://github.com/ant-design/ant-design/issues/51627